internal/observability/event/context.go

```diff
@@ -267,28 +266,61 @@ func WriteSysEvent(ctx context.Context, caller Op, data map[string]interface{})
 		}
 	}
 
-	eventer := SysEventer()
-	if eventer == nil {
-		logger := hclog.New(nil)
-		logger.Error(fmt.Sprintf("%s: no eventer available to write sysevent: (%s) %+v", op, caller, data))
-		return
+	eventer, ok := EventerFromContext(ctx)
+	if !ok {
+		eventer = SysEventer()
+		if eventer == nil {
+			logger := hclog.New(nil)
+			logger.Error(fmt.Sprintf("%s: no eventer available to write sysevent: (%s) %+v", op, caller, info))
+			return
+		}
 	}
 
 	id, err := newId(string(SystemType))
 	if err != nil {
 		eventer.logger.Error(fmt.Sprintf("%s: %v", op, err))
-		eventer.logger.Error(fmt.Sprintf("%s: unable to generate id while writing sysevent: (%s) %+v", op, caller, data))
+		eventer.logger.Error(fmt.Sprintf("%s: unable to generate id while writing sysevent: (%s) %+v", op, caller, info))
 	}
 
 	e := &sysEvent{
 		Id:      Id(id),
 		Version: sysVersion,
 		Op:      caller,
-		Data:    data,
+		Data:    info,
 	}
 	if err := eventer.writeSysEvent(ctx, e); err != nil {
 		eventer.logger.Error(fmt.Sprintf("%s: %v", op, err))
 		eventer.logger.Error(fmt.Sprintf("%s: unable to write sysevent: (%s) %+v", op, caller, e))
 		return
 	}
 }
+
+// MissingKey defines a key to be used as the "missing key" when ConvertArgs has
+// an odd number of args (it's missing a key in its key/value pairs)
+const MissingKey = "EXTRA_VALUE_AT_END"
+
+// ConvertArgs will convert the key/value pair args to a map.  If the args
+// provided are an odd number (they're missing a key in their key/value pairs)
+// then MissingKey is used to the missing key.
```

Comment:
Interesting, I think I prefer the previous pattern with the `map[string]interface` (and the short hand types of `I`, `D`, H`). This approach of taking `args ...interface{}` for something that needs to be a map seems error prone. With the `map[string]interface` it is enforced at compile time by the type. With this approach you either need to have an error, panic, or this `MissingKey` approach which would likely not be caught until much later after a bug is introduced.

Reply:
Honestly, I like the `args ...interface{}` better and it aligns with the existing hclog pattern most devs coming to boundary will be familiar with for sort of key/value set of parameters

Reply:
I do agree that there's more chance for a dev to not pass in key/value but I also don't think it's a huge problem if they do.  The data will still find its way into an event.

Reply:
I agree with the concerns of @tmessi , but I am a fan of the `args ...interface{}` and am happy with how this PR handles MissingKey is correct.

Reply:
The `MissingKey` logic assumes that:

1. It is a key that is missing, and not a value.
2. The missing key is the last k/v pair

I think it is equally possible that a value is missing, or that a key/value is missing somewhere in the middle of the args, which would mess up all of the following pairs.

While this may not be the most critical part, these kinds of errors are non-issues with the `map[string]interface{}` since it is compile time.

Reply:
True, it is clearly a best effort solution, either way it should be fixed once detected (whether key or value). However, as @jimlambrt pointed out this is the existing pattern in `hclog` and the pattern Boundary is currently using for all logging. We could also consider adding a custom vet in the future to catch this during CI, something similar to what hclog has: https://github.com/hashicorp/go-hclog/tree/master/hclogvet

Reply:
@tmessi I understand the tradeoffs and assumptions in the proposed solution that uses `args... interface{}` when it comes to divining the callers intent if the key/value pairs have an off-by-one problem.